### PR TITLE
[MM-58159] Add admin setting for notification monitoring alongside feature flag

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
Adding a setting to accompany the feature for Notification Metrics, so that admins can turn it off via the System Console if they choose. Setting is default-on.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-58159

```release-note
Added a setting for Notification Metrics `MetricsSettings.EnableNotificationMetrics`
```

### CODE REVIEW

**Review by M-ZubairAhmed — APPROVED** [review]

File: server/channels/app/notification.go
Comment (larkox) [inline server/channels/app/notification.go:1]: I would prefer some util function `AreNotificationMetricsEnabled` (or something similar) that do this calculation for us. Not sure where should that function live (a method of any of the objects, or a util function that receives the config) but would clean up part of this.

**Review by larkox — APPROVED** [review]
LGTM, just a improvement proposal.

**Comment by cwarnermm** [comment]
@devinbinnie - Is this new admin config setting _behind_ a feature flag?

**Comment by devinbinnie** [comment]
> @devinbinnie - Is this new admin config setting _behind_ a feature flag?

Yep! It would be `NotificationMonitoring` or `MM_FEATUREFLAGS_NOTIFICATIONMONITORING` for the env variable.

**Pull request merged**